frontend/src/metabase/entities/actions/actions.ts

```diff
@@ -201,6 +207,50 @@ const Actions = createEntity({
   actions: {
     enableImplicitActionsForModel,
   },
+  objectActions: {
+    createPublicLink: createAction(
+      CREATE_PUBLIC_LINK,
+      ({ id }: { id: WritebackActionId }) => {
+        return ActionsApi.createPublicLink({ id }).then(
+          ({ uuid }: { uuid: string }) => {
+            return {
+              id,
+              uuid,
+            };
+          },
+        );
```

Comment:
Do you think it'd be possible to do an API call and then dispatch `Actions.actionTypes.update`, so we don't need to extend the entity reducer?

Reply:
I was trying to do that, but didn't know how to do it.

Reply:
@kulyk Seems `Entity.actionTypes.UPDATE` expects the payload to be normalized by `normalizr` which we don't necessary have to do in this case. It would be indirect in my opinion to try to reuse that action since we need to try to construct a payload that's normalized by `normalizr` manually which doesn't sound too enticing to me.
<img width="724" alt="image" src="https://user-images.githubusercontent.com/1937582/215120705-7d0a636f-b5a1-4b1a-8c1d-90e7f9e22aad.png">

I honestly expect we expose some kind of action that would update the entity automatically by passing it partial properties that are changed, but `*.actions.update` is currently reserved for calling the endpoint that seem there's no way to tell the action "Hey I just want to update the store" yet.

https://github.com/metabase/metabase/blob/a243b16cca8b425c303961d95429752997db4641/frontend/src/metabase/lib/entities.js#L257-L259

I feel like this would require more tinkering, what do you think about leaving it as is for now?

Reply:
Yeah, let's keep things as is if it's not trivial